main/main.cpp

```diff
@@ -4141,15 +4142,33 @@ bool Main::iteration() {
 	frames++;
 	Engine::get_singleton()->_process_frames++;
 
+	// Determine the frame that took the most time to render in the last 100 rendered frames.
+	recent_frametimes[frames % 100] = ticks_elapsed;
```

Comment:
'frames' is reset to 0 every second (line 4186), so you won't be overwriting all recent frametimes if the FPS is below 100.